Ruddarr/Views/Shared/CachedAsyncImage.swift

```diff
@@ -0,0 +1,91 @@
+import Nuke
+import NukeUI
+
+import SwiftUI
+
+struct CachedAsyncImage: View {
+    let url: String?
+
+    var body: some View {
+        if url == nil {
+            PlaceholderImage(icon: "text.below.photo")
+        } else {
+            LazyImage(request: imageRequest(url)) { state in
+                if let image = state.image {
+                    image.resizable()
+                } else if state.error != nil {
+                    let _ = print("CachedAsyncImage: \(state.error!)")
+                    PlaceholderImage(icon: "network.slash")
+                } else {
+                    PlaceholderImage(icon: "text.below.photo")
+                }
+            }.pipeline(imagePipeline())
+        }
+    }
+    
+    func imagePipeline() -> ImagePipeline {
+        var config = ImagePipeline.Configuration.withDataCache(
+            name: "com.github.radarr.DataCache"
+        )
+
+        config.dataCachePolicy = .automatic
+            
+        return ImagePipeline(configuration: config)
+    }
+
+    func imageRequest(_ urlString: String?) -> ImageRequest {
+        URLSession.shared.configuration.waitsForConnectivity = true
```

Comment:
@pteasima: I'm not sure if this is the correct place to set this?

https://github.com/kean/Nuke/blob/main/Documentation/Nuke.docc/Performance/performance-guide.md#auto-retry

Reply:
@tillkruss I very much a fan of functional programming, so I prefer if any function that returns a value is pure (has no side-effects). It's ok to make side-effects from functions returning Void (as its clear that a side-effect is the only reason for calling such a function in the first place).

As for this case, I'd call it very early in the app lifecycle, e.g. in RuddarrApp.init (I may have complained about side-effects in initilizers before, but app init is a common place to do this and is just a shorthand for setting up a `@UIApplicationDelegateAdaptor` and doing it in `application(_, didFinishLaunchingWithOptions:)` method.

Be aware that this configures the shared URL session singleton, and will affect other request you're making from that same session, e.g. `MovieModel.fetch`. If we don't want that, we should use a separate URLSession instance for at least one of these (Nuke or MovieModel).

Reply:
Do you think we should just define it in `RuddarrApp`, or is it better to use inline explicitly?

Is it possible to give `URLRequest` a `URLSession` or how would I do this inline?